### PR TITLE
HHH-12730 : User types built using 5.1 are not binary compatible with…

### PR DESCRIPTION
**Pull request by gbadner**
… 5.3

https://hibernate.atlassian.net/browse/HHH-12730

No need to review this if we decide to use compatibility transformer instead.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/type/ProcedureParameterNamedBinder.java [hunk 1]
```diff
@@ -31,7 +32,7 @@
 	 * Bind a value to the JDBC prepared statement, ignoring some columns as dictated by the 'settable' parameter.
 	 * Implementors should handle the possibility of null values.
 	 * Does not support multi-column type
-	 *
+	 *SubTransactionType
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/type/ProcedureParameterNamedBinder.java]: Looks like a copypasto.

**Comment by gbadner** [comment]
This approach won't work, so closing.

**Comment by gbadner** [comment]
Re-opening since it appears to be needed after all.

File: hibernate-core/src/main/java/org/hibernate/type/ProcedureParameterNamedBinder.java (hunk 1 above)
Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/type/ProcedureParameterNamedBinder.java]: Fixed.

**Comment by gsmet** [comment]
@gbadner I amended your commit with a small fix and added an additional one for the other `AbstractPersistentCollection`. If the constructor is needed for `PersistentBag`, I suspected it would be needed by the others.

I will take a look at the transformer and decide if we should remove some methods later today. I suspect the ones in user types called by ORM should be removed.

**Comment by gsmet** [comment]
Pushed a few other fixes and merged.

**Pull request merged**